packages/storage/amplify_storage_s3/example/integration_test/download_data_test.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:convert';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3/amplify_storage_s3.dart';
+import 'package:amplify_storage_s3_example/amplifyconfiguration.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'utils/configure.dart';
+import 'utils/sign_in_new_user.dart';
+import 'utils/tear_down.dart';
+
+void main() {
+  late String bytesPath;
+  late List<int> bytesData;
+  late String identityName;
+  late List<int> identityData;
+  late String userIdentityId;
+  late String metaDataPath;
+  late Map<String, String> metadata;
+
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('downloadData()', () {
+    setUpAll(() async {
+      await configure(amplifyEnvironments['main']!);
+      userIdentityId = await signInNewUser();
+
+      bytesPath = 'public/download-data-${uuid()}';
+      bytesData = 'test data'.codeUnits;
+
+      await Amplify.Storage.uploadData(
+        path: StoragePath.fromString(bytesPath),
+        data: HttpPayload.bytes(bytesData),
+      ).result;
+
+      identityData = 'with identity ID'.codeUnits;
+      identityName = 'upload-data-with-identity-id-${uuid()}';
+
+      await Amplify.Storage.uploadData(
+        data: HttpPayload.bytes(identityData),
+        path: StoragePath.fromIdentityId(
+          (identityId) => 'private/$identityId/$identityName',
+        ),
+      ).result;
+
+      metaDataPath = 'public/download-data-get-properties-${uuid()}';
+
+      metadata = {'description': 'foo'};
+
+      await Amplify.Storage.uploadData(
+        path: StoragePath.fromString(metaDataPath),
+        data: HttpPayload.bytes('get properties'.codeUnits),
+        options: StorageUploadDataOptions(
+          pluginOptions: const S3UploadDataPluginOptions(
+            getProperties: true,
+          ),
+          metadata: metadata,
+        ),
+      ).result;
+
+      addTearDownPaths(
+        [
+          StoragePath.fromString(bytesPath),
+          StoragePath.fromIdentityId(
+            (identityId) => 'private/$identityId/$identityName',
+          ),
+        ],
+      );
+    });
+
+    group('with options', () {
+      testWidgets('from identity ID', (_) async {
```

Comment:
*nit:* I would move this test out of this group. Using a path with an IdentityId is not a plugin option so it doesn't really belong in the "with options" group IMO.